test/unit/serverless/aws-lambda.test.js

```diff
@@ -1174,219 +1209,230 @@ tap.test('AwsLambda.patchLambdaHandler', (t) => {
+        assert.equal(transaction.type, 'bg')
+        assert.equal(transaction.getFullName(), expectedBgTransactionName)
+        assert.ok(transaction.isActive())
 
         throw error
       })
     })
 
     wrappedHandler(stubEvent, stubContext, stubCallback)
       .then(() => {
-        t.error(new Error('wrapped handler should fail and go to catch block'))
-        t.end()
+        end(Error('wrapped handler should fail and go to catch block'))
       })
       .catch((err) => {
-        t.equal(err, error)
-        t.equal(transaction.isActive(), false)
+        assert.equal(err, error)
+        assert.equal(transaction.isActive(), false)
 
-        t.end()
+        end()
       })
   })
 
-  t.test(
+  await t.test(
     'should record error event when func is async an UnhandledPromiseRejection is thrown',
-    (t) => {
+    (t, end) => {
+      const { agent, awsLambda, error, stubEvent, stubContext, stubCallback } = t.nr
       agent.on('harvestStarted', function confirmErrorCapture() {
         const errors = agent.errors.traceAggregator.errors
-        t.equal(errors.length, 1)
+        assert.equal(errors.length, 1)
 
         const noticedError = errors[0]
         const [, transactionName, message, type] = noticedError
-        t.equal(transactionName, expectedBgTransactionName)
-        t.equal(message, errorMessage)
-        t.equal(type, 'SyntaxError')
+        assert.equal(transactionName, expectedBgTransactionName)
+        assert.equal(message, errorMessage)
+        assert.equal(type, 'SyntaxError')
       })
 
       let transaction
       const wrappedHandler = awsLambda.patchLambdaHandler(async () => {
         transaction = agent.tracer.getTransaction()
         // eslint-disable-next-line no-new
         new Promise(() => {
-          t.ok(transaction)
-          t.equal(transaction.type, 'bg')
-          t.equal(transaction.getFullName(), expectedBgTransactionName)
-          t.ok(transaction.isActive())
+          assert.ok(transaction)
+          assert.equal(transaction.type, 'bg')
+          assert.equal(transaction.getFullName(), expectedBgTransactionName)
+          assert.ok(transaction.isActive())
 
           throw error
         })
 
         await new Promise((resolve) => setTimeout(resolve, 1))
       })
 
-      process.on('unhandledRejection', (err) => {
-        t.equal(err, error)
-        t.equal(transaction.isActive(), false)
-
-        t.end()
+      tempOverrideUncaught({
+        t,
+        type: tempOverrideUncaught.REJECTION,
+        handler(err) {
+          assert.equal(err, error)
+          assert.equal(transaction.isActive(), false)
+          end()
+        }
       })
 
       wrappedHandler(stubEvent, stubContext, stubCallback)
     }
   )
 
-  t.test('should record error event when error is thrown', (t) => {
-    helper.temporarilyOverrideTapUncaughtBehavior(tap, t)
+  await t.test('should record error event when error is thrown', (t, end) => {
+    const plan = tspl(t, { plan: 8 })
```

Comment:
nice. i saw `node:test` has this but not until 22. I ended up just refactoring away some of the plans as i felt they were unnecessary but good to know we can convert to use this